nano/nano_rpc/entry.cpp

```diff
@@ -0,0 +1,139 @@
+#include <boost/lexical_cast.hpp>
+#include <boost/log/expressions.hpp>
+#include <boost/log/utility/setup/common_attributes.hpp>
+#include <boost/log/utility/setup/file.hpp>
+#include <boost/program_options.hpp>
+#include <nano/lib/errors.hpp>
+#include <nano/lib/jsonconfig.hpp>
+#include <nano/lib/utility.hpp>
+#include <nano/nano_wallet/icon.hpp>
+#include <nano/node/cli.hpp>
+#include <nano/node/ipc.hpp>
+#include <nano/node/working.hpp>
+#include <nano/rpc/rpc.hpp>
+#include <nano/rpc/rpcrequestprocessor.hpp>
+
+namespace
+{
+void logging_init (boost::filesystem::path const & application_path_a)
+{
+	static std::atomic_flag logging_already_added = ATOMIC_FLAG_INIT;
+	if (!logging_already_added.test_and_set ())
+	{
+		boost::log::add_common_attributes ();
+		auto path = application_path_a / "log";
+
+		uintmax_t max_size{ 128 * 1024 * 1024 };
+		uintmax_t rotation_size{ 4 * 1024 * 1024 };
+		bool flush{ true };
+		boost::log::add_file_log (boost::log::keywords::target = path, boost::log::keywords::file_name = path / "rpc_log_%Y-%m-%d_%H-%M-%S.%N.log", boost::log::keywords::rotation_size = rotation_size, boost::log::keywords::auto_flush = flush, boost::log::keywords::scan_method = boost::log::sinks::file::scan_method::scan_matching, boost::log::keywords::max_size = max_size, boost::log::keywords::format = "[%TimeStamp%]: %Message%");
+	}
+}
+
+void run (boost::filesystem::path const & data_path)
+{
+	boost::filesystem::create_directories (data_path);
+	boost::system::error_code error_chmod;
+	nano::set_secure_perm_directory (data_path, error_chmod);
+	std::unique_ptr<nano::thread_runner> runner;
+
+	nano::rpc_config rpc_config;
+	auto error = nano::read_and_update_rpc_config (data_path, rpc_config);
+	if (!error)
+	{
+		logging_init (data_path);
+		boost::asio::io_context io_ctx;
+		try
+		{
+			nano::ipc_rpc_processor ipc_rpc_processor (io_ctx, rpc_config);
+			auto rpc = nano::get_rpc (io_ctx, rpc_config, ipc_rpc_processor);
+			rpc->start ();
+			runner = std::make_unique<nano::thread_runner> (io_ctx, rpc_config.io_threads);
+			runner->join ();
+		}
+		catch (const std::runtime_error & e)
+		{
+			std::cerr << "Error while running rpc (" << e.what () << ")\n";
+		}
+	}
+	else
+	{
+		std::cerr << "Error deserializing config: " << error.get_message () << std::endl;
+	}
+}
+}
+
+int main (int argc, char * const * argv)
+{
+	nano::set_umask ();
+
+	boost::program_options::options_description description ("Command line options");
+
+	// clang-format off
+	description.add_options ()
+		("help", "Print out options")
+		("version", "Prints out version")
+		("daemon", "Start node daemon")
```

Comment:
Should probably say "Start RPC deamon" so people don't think this will start a node when you start the rpc process.

Reply:
yep :D, thanks.